### PR TITLE
DTL enforcement

### PR DESCRIPTION
**Pull request by cccs-douglass**

### CODE REVIEW

File: assemblyline_core/submission_client.py [hunk 1]
```diff
@@ -133,10 +134,17 @@ def submit(self, submission_obj: SubmissionObject, local_files: List = None, com
...
-        else:
-            expiry = None
+        # Figure out the expiry for the submission if none was provided
+        if expiry is None:
+            if submission_obj.params.ttl:
+                expiry = epoch_to_iso(submission_obj.time.timestamp() + submission_obj.params.ttl * 24 * 60 * 60)
+
+        # Enforce the max_dtl
+        if self.config.submission.max_dtl > 0:
+            max_expiry = now_as_iso(self.config.submission.max_dtl)
```
Comment (cccs-sgaron) [inline assemblyline_core/submission_client.py:144]: Shouldn't we multiply max_dtl by 24*60*60 ?

File: assemblyline_core/submission_client.py [hunk 2]
```diff
@@ -232,7 +240,7 @@ def _ready_file(self, local_path: str, expiry, classification) -> Tuple[str, int
                 local_path = extracted_path
 
             self.datastore.save_or_freshen_file(fileinfo['sha256'], fileinfo, expiry,
-                                                al_meta['classification'], redis=self.redis)
```
Comment (cccs-sgaron) [inline assemblyline_core/submission_client.py:235]: Why did we stop passing the redis connection?

File: assemblyline_core/submission_client.py (hunk 1 above)
Comment (cccs-douglass) [reply assemblyline_core/submission_client.py:144]: yes we should

File: assemblyline_core/submission_client.py (hunk 2 above)
Comment (cccs-douglass) [reply assemblyline_core/submission_client.py:235]: it doesn't use it any more since we switched from redis locks to elastic versioning

**Review by cccs-sgaron — APPROVED** [review]

**Pull request merged**